.github/workflows/bun.yml

```
@@ -0,0 +1,49 @@
+name: DD Bun CI
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: oven-sh/setup-bun@v2
+    - uses: actions/checkout@v4
+    - run: bun i
+    - run: bun run format:check
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: oven-sh/setup-bun@v2
+    - uses: actions/checkout@v4
+    - run: bun i
+    - run: bun run lint:check
```

I think we could add names to these steps for nicer display when checking run logs:
```suggestion
    - name: Install
      run: bun i
    - name: Lint
      run: bun run lint:check
```

Same for the rest of the checks using `run`.